applications/serial_lte_modem/src/gnss/slm_at_gnss.c

```diff
@@ -774,9 +776,123 @@ static void on_cloud_evt_cell_pos_data_received(const struct nrf_cloud_data *con
 	}
 }
 
+static void cloud_cmd_wk(struct k_work *work)
+{
+	int ret;
+	char *cmd_rsp;
+
+	ARG_UNUSED(work);
+
+	if (strcmp(cloud_command, "DEVICE") == 0) {
+		(void)nrf_cloud_disconnect();
+		return;
+	} /* else cloud_command == "MODEM" */
+
+	/* Send AT command to modem */
+	ret = nrf_modem_at_cmd(rsp_buf, sizeof(rsp_buf), "%s\r", rsp_buf);
+	if (ret < 0) {
+		LOG_ERR("AT command failed: %d", ret);
+		return;
+	} else if (ret > 0) {
+		LOG_WRN("AT command error, type: %d", nrf_modem_at_err_type(ret));
+	}
+	LOG_INF("MODEM RSP %s", rsp_buf);
+	/* replace " with ' for JSon string-type */
+	for (int i = 0; i < strlen(rsp_buf); i++) {
+		if (rsp_buf[i] == '\"') {
+			rsp_buf[i] = '\'';
+		}
+	}
+	/* format JSon reply */
+	cmd_rsp = k_malloc(strlen(rsp_buf) + 64);
```

Comment:
to get rid of the magic `64`, you could:
`#define RSP_JSON_TEMPLATE "{\"appId\":\"MODEM\", \"messageType\":\"RSP\", \"data\":\"%s\"}"`

and then:
`cmd_rsp = k_malloc(strlen(rsp_buf) + sizeof(RSP_JSON_TEMPLATE));`

Reply:
Tried to use a template in the sprintf() next, seems not good
Will try again, if good then allocate buffer as suggested, too